flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/translators/TwoInputTransformationTranslator.java

```diff
@@ -45,7 +46,11 @@
 		boolean isKeyed =
 			transformation.getStateKeySelector1() != null && transformation.getStateKeySelector2() != null;
 		if (isKeyed) {
```

Comment:
Similar to the `MultiInput` shall we account for the situation if only one input is keyed?

Reply:
I went with the current behaviour, but yes, we could do that. So in that case the non-keyed input would be processed before all the keyed input. Same for the general multi-input.

Reply:
Yes, I'd do it that way. Otherwise I'd proceed with https://github.com/apache/flink/pull/14324. The current state leads to possibly corrupted results.

Reply:
I'll do that then. 👌